### PR TITLE
fix: save selected theme

### LINKED ISSUE #673: Unintened Switching of Themes in WebServer
### Description
It seems our web server's default theme is set to dark but whenever I set my theme to **light** and then enter a challenge the theme switches back to **dark**.

### What is the expected behavior?
To continue with the white theme.

### What is the actual behavior?
It switched me back to the Dark theme.

### Is the bug confirmed?
**Yes** (By @commjoen )

### Related images
Screenshot 2023-03-08 054842

Screenshot 2023-03-08 052550

### PR DESCRIPTION
**Pull request by turjoc120**
### What kind of changes does this PR include?

-   [x] Fixes or refactors
-   [ ] A new challenge
-   [ ] Additional documentation
-   [ ] Something else

#### Description
closes #673 

### Checklist:

-   [ ] All the contributions made are solely the work of me and my co-authors
-   [ ] I tested the changes in this PR (if applicable)
-   [ ] I added unit tests to ensure my change works (when change in Java or on front-end code)
-   [ ] The PR passes pre-commit hooks and automated tests

### CODE REVIEW

File: src/main/resources/static/theme-toggle.js [hunk 1]
```diff
@@ -1,18 +1,12 @@
 (function () {
-  const darkModeMediaQuery = window.matchMedia('(prefers-color-scheme: dark)')
```
[the remark this answers is not in the document]
Comment (bendehaan) [reply src/main/resources/static/theme-toggle.js:2]: How about using this user preference as default initial setting, saving that to local storage?

**Review by bendehaan — CHANGES_REQUESTED** [review]
It looks like the user default prefrence now always overrides the user setting for the site?

**Comment by commjoen** [comment]
Maybe it is better to capture this as a function with a jest test :-)?

**Comment by bendehaan** [comment]
Hi @turjoc120, are you currently working on this fix, or planning to continue work anytime soon?

File: src/main/resources/static/theme-toggle.js [hunk 2]
```diff
@@ -1,11 +1,14 @@
 (function () {
   const darkModeMediaQuery = window.matchMedia('(prefers-color-scheme: dark)')
 
+  const initialTheme = darkModeMediaQuery.matches || localStorage.getItem('darkMode') === 'true'
```
Comment (commjoen) [inline src/main/resources/static/theme-toggle.js:4]:
```suggestion
  const initialTheme = localStorage.getItem('darkMode') === 'true'
```
This fixed it for me :)

**Review by bendehaan — APPROVED** [review]

**Pull request merged**